engine/table/src/main/java/io/deephaven/engine/table/impl/select/SelectColumn.java

```diff
@@ -178,36 +189,41 @@ default void validateSafeForRefresh(final BaseTable<?> sourceTable) {
      */
     SelectColumn copy();
 
-    class ExpressionAdapter implements Expression.Visitor, Value.Visitor {
+    class ExpressionAdapter implements Expression.Visitor<SelectColumn> {
         private final ColumnName lhs;
-        private SelectColumn out;
 
         ExpressionAdapter(ColumnName lhs) {
             this.lhs = Objects.requireNonNull(lhs);
         }
 
-        public SelectColumn getOut() {
-            return Objects.requireNonNull(out);
+        @Override
+        public SelectColumn visit(ColumnName rhs) {
+            return new SourceColumn(rhs.name(), lhs.name());
+        }
+
+        @Override
+        public SelectColumn visit(Literal rhs) {
+            return SelectColumnFactory.getExpression(String.format("%s=%s", lhs.name(), Strings.of(rhs)));
         }
 
         @Override
-        public void visit(Value rhs) {
-            rhs.walk((Value.Visitor) this);
+        public SelectColumn visit(Filter rhs) {
+            return SelectColumnFactory.getExpression(String.format("%s=%s", lhs.name(), Strings.of(rhs)));
```

Comment:
It's not actually identical - the `Strings.of(...)` calls are able to be more efficient when they are broken out like this since they have more explicit type information.

```java
        @Override
        public SelectColumn visit(Literal rhs) {
            return SelectColumnFactory.getExpression(String.format("%s=%s", lhs.name(), Strings.of(rhs)));
        }

        @Override
        public SelectColumn visit(Filter rhs) {
            return SelectColumnFactory.getExpression(String.format("%s=%s", lhs.name(), Strings.of(rhs)));
        }
        ...
```

`io.deephaven.api.Strings#of(io.deephaven.api.literal.Literal)` vs `io.deephaven.api.Strings#of(io.deephaven.api.filter.Filter)`

Reply:
And we are already doing the visitor pattern for the optimized ColumnName case - I'll add a note in the code. This is a pattern I use elsewhere to, and will try and be proactive about putting this sort of note on them if you think it's valuable:

```java
        // Note: it might be tempting to consolidate all of the following getExpression calls to a common function, but
        // then we'd be losing the type information that allows us to call the more explicitly typed Strings#of(<type>)
        // methods.
```

Reply:
I think possibly you didn't notice my specific suggestion to keep the Strings.of part in my original suggestion.

Reply:
AH! Yes, I can simplify it in that regard, will do.